### ISSUE TITLE
Document how to setup and run docker on the mac for development

### ISSUE THREAD

**Issue by rkuris** [ava-labs/firewood#426]
Ideally we use the same setup. Currently I'm using a separate VM for this.

**Comment by richardpringle** [comment]
TL;DR (putting this here for now, will make the proper docs though)

- Setup a dev-environment using your repo-directory as the project directory [^1].
- Configure Docker-Desktop to use `VirtioFS`
- Move your `CARGO_TARGET_DIR` to somewhere that **is not** source-mapped onto the host. 

Then do all the normal Linux stuff. Download `rustup` to get cargo, etc. Set up your build-essentials. These will not be things that I will add to the doc, instead, whoever is setting their stuff up for the first time should add as they see fit. I don't remember all the different things I had to do, but they are no different than setting up any Linux environment. 

If you are seeing build failures, there's a concurrency issue with building onto mapped volumes. You need the target-dir to only live in the container. On your host machine, you don't need to change the behaviour. For step by step debugging, you'll need to use a remote-vs-code instance that you can start from docker-desktope. Otherwise, I do all my development from the host machine, and keep a terminal open for running the entire test suite in the container. Most of the `Merkle` tests, etc. can run fine in the mac environment. 

[^1]: https://docs.docker.com/desktop/dev-environments/

**Comment by richardpringle** [comment]
![image](https://github.com/ava-labs/firewood/assets/3286504/83d6b66d-19e3-4b59-bc73-f67cf68d7329)

**Comment by richardpringle** [comment]
![image](https://github.com/ava-labs/firewood/assets/3286504/2f573c42-10df-4af3-b253-2522820b3ce6)

**Comment by richardpringle** [comment]
![image](https://github.com/ava-labs/firewood/assets/3286504/88c981cb-42b9-4b99-acec-fbca31cca652)
